packages/devtools_app/lib/src/app.dart

```diff
@@ -129,27 +129,29 @@ class DevToolsAppState extends State<DevToolsApp> {
         if (params['uri']?.isNotEmpty ?? false) {
           final embed = params['embed'] == 'true';
           final page = params['page'];
-          final tabs = embed && page != null
-              ? _visibleScreens().where((p) => p.screenId == page).toList()
-              : _visibleScreens();
           return Initializer(
             url: params['uri'],
             allowConnectionScreenOnDisconnect: !embed,
-            builder: (_) => _providedControllers(
-              child: DevToolsScaffold(
-                embed: embed,
-                initialPage: page,
-                tabs: tabs,
-                actions: [
-                  if (serviceManager.connectedApp.isFlutterAppNow) ...[
-                    HotReloadButton(),
-                    HotRestartButton(),
+            builder: (_) {
+              final tabs = embed && page != null
+                  ? _visibleScreens().where((p) => p.screenId == page).toList()
```

Comment:
if you try to open an embedded page on an app that doesn't support it we should display an error message explaining what went wrong.

Reply:
I've added a TODO for this and will do it it a separate PR to avoid adding more to this one (it's already grown a fair bit!).

Reply:
Opened https://github.com/flutter/devtools/pull/2207